### ISSUE TITLE
Add a DEB repository for tenv

### ISSUE THREAD

**Issue by samuel-phan** [tofuutils/tenv#152]
**Is your feature request related to a problem? Please describe.**
By using a DEB repo, Debian/Ubuntu users can benefit from auto-update like any other package. It's simpler and easier for day-to-day operations.

**Describe the solution you'd like**
Create a PPA repo for tenv, and for every build, add the new version there.

**Describe alternatives you've considered**
N/A.

**Additional context**
N/A.

**Comment by kvendingoldo** [comment]
Good idea. I'm still working on snap in scope of https://github.com/tofuutils/tenv/issues/53, maybe deb should be before it.

**Comment by kvendingoldo** [comment]
Created PPA (https://launchpad.net/tenv), will add publish to it soon.

**Comment by kvendingoldo** [comment]
Created a basic pipeline: https://github.com/tofuutils/tenv/commit/ea91aee5020e2e39f55661559cff654719a67fec
will check how it will works

**Comment by kvendingoldo** [comment]
example: https://github.com/diogenes1oliveira/hashicorp-ppa/tree/master

**Comment by kvendingoldo** [comment]
example 2:https://github.com/mfinelli/ppa

**Comment by samuel-phan** [comment]
@kvendingoldo Any progress on this?

**Comment by kvendingoldo** [comment]
@samuel-phan working on it. Doesn't snap work for u?

**Comment by samuel-phan** [comment]
Thanks.

On Linux Mint, `snap` is not installed by default, so I don't use it. I'm always a bit reluctant to use tools in a sandbox, because it might block some features that needs bindings with other parts on the OS libs (eg custom under development TF providers).

**Comment by kvendingoldo** [comment]
Ok. I will try to deliver deb repo this week

**Comment by samuel-phan** [comment]
As long as it's publicly possible to use `apt` to install tenv on Ubuntu and get updates just by using `apt update && apt upgrade`, it's fine by me.

**Comment by kvendingoldo** [comment]
As I understand correctly, you have to do:

```
curl -1sLf \
  'https://dl.cloudsmith.io/public/tofuutils/tenv/setup.deb.sh' \
  | sudo -E bash
apt-get install tenv=4.2.4
```

**Comment by kvendingoldo** [comment]
Looks like, in such a case, the best case is downloading deb from our release on GitHub (e.g.: https://github.com/tofuutils/tenv/releases/tag/v4.2.4). We'll try to add ourselves to the official registry that will allow users to use just apt-get install, but it's a long process that we definitely will do.

**Comment by julian-alarcon** [comment]
I also don't like installation using .sh files @samuel-phan .

I was able to setup the repository without using the .sh checking that was doing the installation file.

You can find the needed information using this URL:
https://dl.cloudsmith.io/public/tofuutils/tenv/config.deb.txt?distro=ubuntu&codename=noble&version=24.04

This will download an old apt source file, with that information as I prefer new apt DEB822 format, and I'm using Ubuntu 24.04, then I needed to:

Add this information to a new file  `/etc/apt/sources.list.d/tenv-ubuntu-repo-noble.sources`

```sh
Architectures: amd64
Components: main
X-Repolib-Name: tenv-ubuntu-repo-noble
Signed-By: /etc/apt/keyrings/tenv-ubuntu-repo-noble.asc
Suites: noble
Types: deb
URIs: https://dl.cloudsmith.io/public/tofuutils/tenv/deb/ubuntu
```

Also, related to the gpg key, I check the .sh file and the current origin is this: https://dl.cloudsmith.io/public/tofuutils/tenv/gpg.8ACD4386ADD982F6.key , So i used that file, renamed it and moved it to:
`/etc/apt/keyrings/tenv-ubuntu-repo-noble.asc` 

SHA512 of .key: `ebf534b6e644678a0c21806426deb56d57ca7e6297881b2c66c5e60c7fe66937a266982e07d556339292e3447ad264aba98af5f61bfde727befbba3fc016f70f`

Now I was able to do a normal installation.

BTW I automated this with ansible in my local, if you want to check the Ansible role (it's not complex)
https://gitlab.com/julian-alarcon/linux-setup/-/tree/main/roles/tenv?ref_type=heads

**Comment by julian-alarcon** [comment]
As the repository exists (it's just a matter of form on how to install it from APT repo), this should be closed.

**Comment by julian-alarcon** [comment]
I also found a guide from cloudsmith to manually setup the repo: https://cloudsmith.io/~tofuutils/repos/tenv/setup/#debian-repository-setup
